gpii/node_modules/deviceReporter/src/DeviceReporter.js

```diff
@@ -32,21 +32,85 @@ fluid.defaults("gpii.deviceReporter", {
             type: "get"
         }
     },
+    components: {
+        platformReporter: {
+            type: "gpii.platformReporter.native"
+        }
+    },
+    invokers: {
+        fireResponse: {
+            funcName: "gpii.deviceReporter.fireResponse",
+            args: ["{arguments}.0", "{arguments}.1", "{platformReporter}"]
+        }
+    }
+});
+
+gpii.deviceReporter.fireResponse = function (requestProxy, installedSolutions, platformReporter) {
+    requestProxy.events.onSuccess.fire({
+        solutions: installedSolutions,
+        OS: platformReporter.reportPlatform()
+    });
+};
+
+fluid.defaults("gpii.deviceReporter.static", {
+    gradeNames: ["autoInit", "gpii.deviceReporter.base"],
     installedSolutionsUrl: "",
     root: path.join(__dirname, ".."),
     components: {
         installedSolutionsDataSource: {
             type: "kettle.dataSource.URL",
             options: {
-                url: "{gpii.deviceReporter}.options.installedSolutionsUrl"
+                url: "{gpii.deviceReporter.static}.options.installedSolutionsUrl"
             }
+        }
+    },
+    invokers: {
+        get: {
+            funcName: "gpii.deviceReporter.static.get",
+            args: ["{arguments}.0", "{deviceReporter}"]
+        }
+    }
+});
+
+gpii.deviceReporter["static"].get = function (requestProxy, deviceReporter) {
+    deviceReporter.installedSolutionsDataSource.get(null, function onSuccess(solutions) {
+        deviceReporter.fireResponse(requestProxy, solutions);
+    });
+};
+
+fluid.defaults("gpii.deviceReporter.live", {
+    gradeNames: ["autoInit", "gpii.deviceReporter.base"],
+    invokers: {
+        get: {
+            funcName: "gpii.deviceReporter.live.get",
+            args: ["{arguments}.0", "{deviceReporter}", "{flowManager}.solutionsRegistryDataSource"]
         },
-        platformReporter: {
-            type: "gpii.platformReporter.native"
+        resolveName: {
```

Comment:
This method is ok but messy - it will be clearer for readers (and for YOU, VANTTUNEZZ!) if you just use the same scheme that we have for settings handlers - that is, just inject the nameResolver as a component of gpii.deviceReporter.base and expect for all users to invoke it from there (rather than make a fresh invoker) - that is. deviceReporter.nameResolver.resolveName